qf/types/requests.proto

```diff
@@ -0,0 +1,172 @@
+syntax = "proto3";
+package types;
+option go_package   = "github.com/quickfeed/quickfeed/qf/types";
+option swift_prefix = "";
+
+import "qf/types/types.proto";
+
+// Message types that are used as requests and responses in API calls.
+
+message SubmissionLink {
```

Comment:
This should be in the `types.proto` file (it is only used by the database ; not used in `qf.proto`).

Reply:
Hmm, I thought the idea was to have messages saved directly to the database in a separate file (types.proto) and group the rest in the `requests.proto`. `SubmissionLink` is never saved to the database.

On the other hand, I also put `Courses`, `Enrollments' and others in `types.proto` while they also never go to the database. So yeah, `SubmissionLink` belongs in the `types.proto`.

Reply:
Moved.

Reply:
I was thinking about the separation more in terms of which messages are used as request messages in the RPCs. But let's review the decision again once we have some more experience with this separation. Maybe we learn something new. 

I know that some of the messages, such as `Courses` and `Enrollments` are used both for RPCs and stored in database... That's desirable. Perhaps we learn that it would make more sense to have another file with messages that are shared between both... For now I think we have at least some good starting point for the separation we wanted...